lib/pbench/cli/server/reindex.py

```diff
@@ -0,0 +1,309 @@
+from collections import defaultdict
+from typing import Optional
+from urllib.parse import urljoin
+
+import click
+import requests
+from sqlalchemy.orm import Query
+
+from pbench.cli import pass_cli_context
+from pbench.cli.server import config_setup, DateParser, Detail, Verify, Watch
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server import BadConfig
+from pbench.server.database.database import Database
+from pbench.server.database.models.datasets import (
+    Dataset,
+    OperationName,
+    OperationState,
+)
+from pbench.server.database.models.index_map import IndexMap
+from pbench.server.database.models.server_settings import (
+    OPTION_SERVER_INDEXING,
+    ServerSetting,
+)
+from pbench.server.sync import Sync
+
+detailer: Optional[Detail] = None
+watcher: Optional[Watch] = None
+verifier: Optional[Verify] = None
+
+
+SELECTORS = set(("since", "until", "id", "name"))
+
+
+def datasets(kwargs) -> Query[Dataset]:
+    """Return a filtered query to select datasets
+
+    NOTE: filters are all AND-ed, so there's no way to select by id OR name.
+    This could be handled in various means that are awkward in click, e.g.
+    with a "mode switch" and callback handling to push the selector option
+    values into "or" or "and" lists. While I could see that being "cool" I
+    don't see enough value to bother with it now.
+
+    Returns:
+        a Filtered SQLAlchemy Query
+    """
+    query = (
+        Database.db_session.query(Dataset)
+        .execution_options(stream_results=True)
+        .yield_per(2000)
+    )
+    filters = []
+    ids = kwargs.get("id")
+    if ids and isinstance(ids, (list, tuple)):
+        verifier.status(f"Filter resource IDs match {ids}")
+        filters.append(Dataset.resource_id.in_(ids))
+    names = kwargs.get("name")
+    if names and isinstance(names, (list, tuple)):
+        verifier.status(f"Filter names match {names}")
+        filters.append(Dataset.name.in_(names))
+    since = kwargs.get("since")
+    until = kwargs.get("until")
+    if since:
+        verifier.status(f"Filter since {since}")
+        filters.append(Dataset.uploaded >= since)
+    if until:
+        verifier.status(f"Filter until {until}")
+        filters.append(Dataset.uploaded <= until)
+    return query.filter(*filters)
+
+
+def summarize_index(dataset: Dataset, kwargs):
+    """A simple Opensearch query to report index statistics
+
+    Args:
+        dataset: a Dataset object
+        kwargs: options
+    """
+    es_url, ca_bundle = kwargs.get("_es")
+    try:
+        indices = list(IndexMap.indices(dataset))
+        if not indices:
+            click.echo(f"{dataset.name} is not indexed")
+            return
+        url = ",".join(indices) + "/_search"
```

Comment:
It doesn't instantiate the list, but it also doesn't remove the local variable, so I don't see any benefit. Same five lines with a few characters shuffled around a bit ...

Reply:
The difference is in the execution:  it constructs the indices string directly instead of constructing a list first and then joining its items.  It's not a big difference, especially because the lists are small, but I thought it was worth considering.

And, of course, we could avoid instantiating the `indices` local variable by combining the `join()` expression with the concatenation, and that would save us a line, but it would also require making the "empty" check more complicated, and I don't think the result would be better.